### PR TITLE
Configure Foundry, Solhint, and Prettier

### PR DESCRIPTION
**Pull request by stevennevins**
This PR creates initial configurations for:
- CI 
- Solhint
- Foundry
- Prettier

### CODE REVIEW

File: lib/forge-std
Comment (xflpt) [inline lib/forge-std:1]: is it mandatory to have it as a submodule ?

File: remappings.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@ensdomains/=node_modules/@ensdomains/
+@ethereum-waffle/=node_modules/@ethereum-waffle/
+@openzeppelin/=node_modules/@openzeppelin/
+@rarible/=node_modules/@rarible/
```
Comment (xflpt) [inline remappings.txt:4]: I actually forgot to remove it, we won't use it anymore. Do you mind including the removal of the @rarible package in the PR ?

File: lib/forge-std
Comment (stevennevins) [reply lib/forge-std:1]: yeah afaik

File: remappings.txt (hunk 1 above)
Comment (stevennevins) [reply remappings.txt:4]: sure np

**Pull request merged**